src/javascripts/ng-admin/Crud/button/maExportToCsvButton.js

```diff
@@ -25,27 +26,78 @@ define(function () {
                         exportFields = listView.fields();
                     }
                     exportView.fields(exportFields);
+                    exportView.filters(listView.filters());
                 }
                 scope.has_export = exportView.fields().length > 0;
                 var formatEntry = entryFormatter.getFormatter(exportView.fields());
 
                 scope.exportToCsv = function () {
+                    var rawEntries;
+                    var nonOptimizedReferencedData;
+                    var optimizedReferencedData;
 
-                    ReadQueries.getAll(exportView, -1, true, $stateParams.search, $stateParams.sortField, $stateParams.sortDir).then(function (response) {
-                        var results = [];
-                        var entries = scope.datastore.mapEntries(scope.entity.name(), exportView.identifier(), exportView.fields(), response.data);
-                        for (var i = entries.length - 1; i >= 0; i--) {
-                            results[i] = formatEntry(entries[i]);
-                        }
-                        var csv = Papa.unparse(results);
-                        var fakeLink = document.createElement('a');
-
-                        fakeLink.setAttribute('href', 'data:application/octet-stream;charset=utf-8,' + encodeURIComponent(csv));
-                        fakeLink.setAttribute('download', scope.entity.name() + '.csv');
-                        fakeLink.click();
-                    }, function (error) {
-                        notification.log(error.message, {addnCls: 'humane-flatty-error'});
-                    });
+                    ReadQueries.getAll(exportView, -1, scope.search(), $stateParams.sortField, $stateParams.sortDir)
```

Comment:
If we use `$stateParams.search` instead of `scope.search()`, it's not the right value (but it should be the same value as in `ListController`...).